docs/src/pages/docs/rag/retrieval.mdx

```diff
@@ -56,19 +56,85 @@ Results include both the text content and a similarity score:
 ]
 ```
 
+For an example of how to use the basic retrieval method, see the [Retrieve Results](../../examples/rag/retrieve-results.mdx) example.
+
 ## Advanced Retrieval options
 
 ### Metadata Filtering
 
-Filter results based on metadata fields to narrow down the search space. This is useful when you have documents from different sources or time periods:
+Filter results based on metadata fields to narrow down the search space. This is useful when you have documents from different sources, time periods, or with specific attributes. Mastra provides a unified MongoDB-style query syntax that works across all supported vector stores.
+
+For detailed information about available operators and syntax, see the [Metadata Filters Reference](/docs/reference/rag/metadata-filters).
+
+Basic filtering examples:
 
 ```ts showLineNumbers copy
+// Simple equality filter
+const results = await pgVector.query("embeddings", embedding, 10, {
+  source: "article1.txt"
+});
+
+// Numeric comparison
+const results = await pgVector.query("embeddings", embedding, 10, {
+  price: { $gt: 100 }
+});
+
+// Multiple conditions
+const results = await pgVector.query("embeddings", embedding, 10, {
+  category: "electronics",
+  price: { $lt: 1000 },
+  inStock: true
+});
+
+// Array operations
 const results = await pgVector.query("embeddings", embedding, 10, {
-  source: "article1.txt",
-  date: { $gt: "2023-01-01" }
+  tags: { $in: ["sale", "new"] }
+});
+
+// Logical operators
+const results = await pgVector.query("embeddings", embedding, 10, {
+  $or: [
+    { category: "electronics" },
+    { category: "accessories" }
+  ],
+  $and: [
+    { price: { $gt: 50 } },
+    { price: { $lt: 200 } }
+  ]
 });
 ``` 
 
+Common use cases for metadata filtering:
+- Filter by document source or type
+- Filter by date ranges
+- Filter by specific categories or tags
+- Filter by numerical ranges (e.g., price, rating)
+- Combine multiple conditions for precise querying
+- Filter by document attributes (e.g., language, author)
+
+For an example of how to use the metadata filtering method, see the [Metadata Filtering](../../examples/rag/filter-rag.mdx) example.
+
+### Vector Query Tool
+
+For more advanced retrieval scenarios, Mastra provides a unified vector query tool that combines semantic search, filtering, and reranking capabilities. This tool is particularly useful when:
```

Comment:
Can we please explicitly say -- "sometimes you want to give your agent a tool that lets your agent query a database. In order to do this...."

Reply:
Whatever our lessons are on tool naming and description for agents around this tool is also probably worth a sentence.